controllers/namespace/namespace_controller.go

```diff
@@ -85,16 +86,22 @@ func (r *ReconcileNamespaces) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, nil
 	}
 
-	var ims dynatracev1alpha1.DynaKubeList
-	if err := r.client.List(ctx, &ims, client.InNamespace(r.namespace)); err != nil {
-		return reconcile.Result{}, fmt.Errorf("failed to query OneAgentIMs: %w", err)
-	}
-
 	var apm dynatracev1alpha1.DynaKube
 	if err := r.client.Get(ctx, client.ObjectKey{Name: oaName, Namespace: r.namespace}, &apm); err != nil {
 		return reconcile.Result{}, fmt.Errorf("failed to query OneAgentAPM: %w", err)
 	}
 
+	tokenName := utils.GetTokensName(&apm)
+	if !apm.Spec.CodeModules.Enabled {
+		r.ensureSecretDeleted(tokenName, targetNS)
+		return reconcile.Result{RequeueAfter: 5 * time.Minute}, nil
+	}
+
+	var ims dynatracev1alpha1.DynaKubeList
+	if err := r.client.List(ctx, &ims, client.InNamespace(r.namespace)); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to query OneAgentIMs: %w", err)
```

Comment:
```suggestion
		return reconcile.Result{}, fmt.Errorf("failed to query OneAgentIMs: %w", err)
```

Please also update `failed to query OneAgentAPM` above.

Reply:
Done